doc/tutorial/sam_file/sam_file_solution2.cpp

```diff
@@ -71,50 +40,47 @@ write_file_dummy_struct go{};
 
 int main()
 {
-    std::filesystem::path tmp_dir = std::filesystem::temp_directory_path(); // get the temp directory
+    std::filesystem::path current_path = std::filesystem::current_path();
 
     // read in reference information
-    seqan3::sequence_file_input reference_file{tmp_dir/"reference.fasta"};
-    std::vector<std::string> ref_ids{};
-    std::vector<seqan3::dna5_vector> ref_seqs{};
+    seqan3::sequence_file_input reference_file{current_path / "reference.fasta"};
+
+    std::vector<std::string> reference_ids{};
+    std::vector<seqan3::dna5_vector> reference_sequences{};
 
     for (auto && record : reference_file)
     {
-        ref_ids.push_back(std::move(record.id()));
-        ref_seqs.push_back(std::move(record.sequence()));
+        reference_ids.push_back(std::move(record.id()));
+        reference_sequences.push_back(std::move(record.sequence()));
     }
 
-    using field_type = seqan3::fields<seqan3::field::id,
-                                      seqan3::field::ref_id,
-                                      seqan3::field::mapq,
-                                      seqan3::field::alignment>;
-
-    seqan3::sam_file_input mapping_file{tmp_dir/"mapping.sam", ref_ids, ref_seqs, field_type{}};
+    // filter out alignments
+    seqan3::sam_file_input mapping_file{current_path / "mapping.sam", reference_ids, reference_sequences};
 
 #if !SEQAN3_WORKAROUND_GCC_93983
-    auto mapq_filter = std::views::filter([] (auto & rec) { return rec.mapping_quality() >= 30; });
+    auto mapq_filter = std::views::filter([] (auto & record) { return record.mapping_quality() >= 30; });
 #endif // !SEQAN3_WORKAROUND_GCC_93983
 
 #if SEQAN3_WORKAROUND_GCC_93983
-    for (auto & [id, ref_id, mapq, alignment] : mapping_file /*| mapq_filter*/)
+    for (auto & record : mapping_file /*| mapq_filter*/)
 #else // ^^^ workaround / no workaround vvv
-    for (auto & [id, ref_id, mapq, alignment] : mapping_file | mapq_filter)
+    for (auto & record : mapping_file | mapq_filter)
 #endif // SEQAN3_WORKAROUND_GCC_93983
```

Comment:
Could we write a comment what the WORKAROUND things mean? As this could be confusing for a newbie.

Reply:
I'll remove it :)

Reply:
There are a couple of these. So I'll remove it in an extra PR.

Reply:
Here: https://github.com/seqan/seqan3/pull/2624